### ISSUE TITLE
Fallback to the default culture resource

### ISSUE THREAD

**Issue by sivashkevych** [hishamco/My.Extensions.Localization.Json#83]
The default implementation of IStringLocalizer returns a string from the default culture resource if the localizer can't find the localization in the current resource.

However, the JsonStringLocalizer returns the localization string key value by default.

I've seen the PR Culture fallback to the rooted resource.
Are you going to merge it?

**Comment by hishamco** [comment]
Seems you mean this PR #41

**Comment by hishamco** [comment]
Seems the PR is too old, I can confirm that culture fallback is working

https://github.com/hishamco/My.Extensions.Localization.Json/blob/3dda94f89f1e74be17d643bae4e8b8291bc617f1/test/My.Extensions.Localization.Json.Tests/JsonStringLocalizerTests.cs#L66-L84

**Comment by sivashkevych** [comment]
Yes. I mean PR [#41](https://github.com/hishamco/My.Extensions.Localization.Json/pull/41).

It seems that culture fallback isn't functioning correctly for the default culture.
In our project, we've included `es.json` and `en-US.json`, setting the default culture to "en-US".

Typically, with the standard _IStringLocalizer_, if a string isn't found in `es.json`, it falls back to `en-US.json`. However, with _JsonStringLocalizer_, it shows the string key instead.

I noticed in the previous PR that you implemented fallback to the default culture.
Merging PR #41 would enable us to have the same fallback behavior as with the original _IStringLocalizer_.

Thank you for considering this.

**Comment by hishamco** [comment]
> Typically, with the standard IStringLocalizer, if a string isn't found in es.json, it falls back to en-US.json. However, with JsonStringLocalizer, it shows the string key instead.

Please show me the localization setup in your code

**Comment by sivashkevych** [comment]
Here is the localization setup:
```csharp
builder.Services.AddJsonLocalization(options => options.ResourcesPath = "JsonResources");
builder.Services.Configure<RequestLocalizationOptions>(options =>
    {
        var supportedCultures = new[]
        {
            new CultureInfo("en-US"),
            new CultureInfo("es"),
        };
        options.DefaultRequestCulture = new RequestCulture("en-US");
        options.SupportedCultures = supportedCultures;
        options.SupportedUICultures = supportedCultures;
        options.ApplyCurrentCultureToResponseHeaders = true;
});

builder.Services.AddMvc()
    .AddViewLocalization()
    .AddDataAnnotationsLocalization();
    
```

**Comment by hishamco** [comment]
As I understand from you the fallback culture in the same culture set is working fine, but the fallback to the default culture is not working, right?

**Comment by sivashkevych** [comment]
Correct. Fallback from `Spain` to `English (US)` doesn't work.

**Comment by hishamco** [comment]
I will create a unit test, and then check

**Comment by sivashkevych** [comment]
> I will create a unit test, and then check

Thank you, sir.

**Comment by hishamco** [comment]
Please check one case before I move further make sure that the requested resource is available in the default culture JSON file in your case `en-US.json` otherwise showing the resource name itself it make sense to me because it's not localized

**Comment by sivashkevych** [comment]
We have a "**PagesHome**" localized string in both `en-US.json` and `es.json`.

I removed it from `es.json`.
After that I can see **_PagesHome_** when I add ?culture=es to the URL
and **_Home_** when I add ?culture=en-US or by default without specifying culture.

**Comment by sivashkevych** [comment]
> Let me double-check or write a functional test for this case

Of course.

**Comment by hishamco** [comment]
@sivashkevych seems you confused with the culture fallback. If the culture is not exists in the supported culture then the fallback will be applied in your case if the culture is `fr` it SHOULD work, but if the localization find the resource on the supported culture it will show the same resource name if the value is not found

Hope it solved your problem

**Comment by hishamco** [comment]
Anything to add here @sivashkevych?
